src/test/java/com/mongodb/hibernate/jdbc/MongoStatementIntegrationTests.java

```diff
@@ -0,0 +1,349 @@
+                                }
+                            ]
+                        }"""
+                        .formatted(COLLECTION_NAME);
+
+        @ParameterizedTest
+        @ValueSource(booleans = {true, false})
+        void testInsert(boolean autoCommit) {
+            var expectedDocs = Set.of(
+                    BsonDocument.parse(
+                            """
+                            {
+                                    _id: 1,
+                                    title: "War and Peace",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                            {
+                                    _id: 2,
+                                    title: "Anna Karenina",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                 }"""),
+                    BsonDocument.parse(
+                            """
+                            {
+                                    _id: 3,
+                                    title: "Resurrection",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 4,
+                                    title: "Crime and Punishment",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 5,
+                                    title: "The Brothers Karamazov",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 6
+                                    title: "Fathers and Sons",
+                                    author: "Ivan Turgenev",
+                                    outOfStock: false
+                                }"""));
+            assertExecuteUpdate(INSERT_MQL, autoCommit, 6, expectedDocs);
+        }
+
+        @ParameterizedTest
+        @ValueSource(booleans = {true, false})
+        void testUpdate(boolean autoCommit) {
+            // given
+            session.doWork(connection -> {
+                connection.setAutoCommit(true);
+                var statement = connection.createStatement();
+                statement.executeUpdate(INSERT_MQL);
+            });
```

Comment:
This piece of code is used twice: in `testUpdate` and `testDelete`. Let's extract it to a `prepareData` method, or something like that.

Reply:
done